specifyweb/frontend/js_src/lib/components/SchemaConfig/Tables.tsx

```diff
@@ -102,10 +117,13 @@ export function TableList({
     return showHiddenTables ? sortedTables : presentFilteredTables;
   }, [showHiddenTables, sortedTables, children]);
 
+  const tablesToMapOver =
+    cacheKey === 'appResources' ? tablesToDisplay : advancedTables;
```

Comment:
this is bad practice and should be avoided if possible
(because needless separate code paths should be avoided if possible, and because cacheKey, as it's name implies, it only used for local storage caching, not for display logic)
you shouldn't need this once the suggestion I added for sortedTables is implemented

Reply:
not really sortedTables and advancedTables do not serve the same purpose. In one case we only want the tables that already have resources in the other we only want the tables that are not advanced

Reply:
Would you have another suggestion than using cacheKey to differentiate the fact that we are in one component or the other?

Reply:
how does this not combine the sortedTables and advancedTables?
```
  const sortedTables = React.useMemo(
    () =>
      allTables
        .filter((table) =>
          filter
            ? filter(showHiddenTables, table)
            : showHiddenTables || !table.isSystem
        )
        .sort(sortFunction(({ name }) => name)),
    [filter, allTables, showHiddenTables]
  );
```

Reply:
For app resources list of table, when the check box is not checked, this is what we want to see: 
![Screenshot 2023-12-22 at 10 57 53 AM](https://github.com/specify/specify7/assets/108160931/21311589-bf85-4351-a362-39452bd00050)
With the code above, this is what is displayed: 
![Screenshot 2023-12-22 at 10 59 51 AM](https://github.com/specify/specify7/assets/108160931/ddd89b43-f9ab-4ca3-856b-c22bbd874882)

Reply:
we still need a check to display sortedTables for schema config and another to display tablesToDisplay for app resources.